### PR TITLE
Thumbnail source can use an icon

### PR DESCRIPTION
**Pull request by alex-page**
### WHY are these changes introduced?

Fixes https://github.com/Shopify/polaris-react/issues/3210

### WHAT is this pull request doing?

Allows icons to be in `Thumbnail` component. This is great for `Dropzone` and other components.

### How to 🎩

Look at updated examples

### 🎩 checklist

* [x] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
* [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
* [x] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
* [x] Updated the component's `README.md` with documentation changes
* [x] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide
* [x] For visual design changes, pinged one of @ HYPD, @ mirualves, @ sarahill, or @ ry5n to update the Polaris UI kit

### CODE REVIEW

File: src/components/Thumbnail/Thumbnail.tsx [hunk 1]
```diff
@@ -25,9 +26,12 @@ export function Thumbnail({source, alt, size = 'medium'}: ThumbnailProps) {
...
-    <span className={className}>
-      <Image alt={alt} source={source} className={styles.Image} />
-    </span>
-  );
+  const content =
+    typeof source === 'string' ? (
+      <Image alt={alt} source={source} />
+    ) : (
+      <Icon accessibilityLabel={alt} source={source} />
+    );
```
Comment (alex-page) [inline src/components/Thumbnail/Thumbnail.tsx:34]: Maybe I should flip this and say `source === 'function'`

File: src/components/Thumbnail/README.md [hunk 2]
```diff
@@ -104,6 +104,14 @@ Use when a thumbnail is a major focal point. Avoid this size in lists of like it
 />
 ```
 
+### Thumbnail with component source
+
+Use when a thumbnail uses a component inside of it.
```
Comment (kyledurand) [inline src/components/Thumbnail/README.md:109]:
```suggestion
Use to render an icon inside of thumbnail.
```

File: src/components/Thumbnail/Thumbnail.scss [hunk 3]
```diff
@@ -13,15 +8,12 @@ $stacking-order: (
   min-width: thumbnail-size(small);
   max-width: 100%;
   border-radius: var(--p-border-radius-base, border-radius());
+  box-shadow: inset 0 0 0 1px rgba(33, 43, 54, 0.1);
```
Comment (kyledurand) [inline src/components/Thumbnail/Thumbnail.scss:11]: Can we use an existing shadow from tokens here?

File: src/components/Thumbnail/Thumbnail.tsx (hunk 1 above)
Comment (kyledurand) [reply src/components/Thumbnail/Thumbnail.tsx:34]: I think this is ok but should we add a couple of tests for it?

**Review by kyledurand — APPROVED** [review]
Could comments but this looks good

File: src/components/Thumbnail/Thumbnail.scss (hunk 3 above)
Comment (alex-page) [reply src/components/Thumbnail/Thumbnail.scss]: I have replaced this with `border('divider')` I can't see any reason it should be an inset line.

File: src/components/Thumbnail/Thumbnail.tsx (hunk 1 above)
Comment (alex-page) [reply src/components/Thumbnail/Thumbnail.tsx:34]: Added some tests.

**Review by kyledurand — APPROVED** [review]

File: src/components/Thumbnail/tests/Thumbnail.test.tsx [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import {mountWithApp} from 'test-utilities/react-testing';
+import {NoteMajor} from '@shopify/polaris-icons';
+
+import {Thumbnail} from '../Thumbnail';
+
+describe('<Thumbnail />', () => {
+  describe('source', () => {
```
Comment (beefchimi) [inline src/components/Thumbnail/tests/Thumbnail.test.tsx:8]: @alex-page / @kyledurand None of these tests have assertions and are therefor causing `type-check` to fail 😢  Does Travis not prevent merging when in-progress/failed?

**Comment by alex-page** [comment]
@beefchimi that is my bad. Fixing it up now.

**Pull request merged**